pandas/sparse/tests/test_sparse.py

```diff
@@ -948,6 +948,23 @@ def setUp(self):
 
         self.empty = SparseDataFrame()
 
+    def test_fill_value_missing_with_binary_operators(self):
+        # GH12723
+        dat1 = np.arange(10, dtype='float')
+        df1 = DataFrame({'foo': dat1}, index=range(10)).to_sparse()
+
+        dat2 = dat1.copy()
+        dat2[5] = np.nan
+        df2 = DataFrame({'foo': dat2}, index=range(10)).to_sparse()
+
+        exp = df2.fillna(0).add(2)
+        res = df2.add(2, fill_value=0)
+        tm.assert_sp_frame_equal(res, exp)
+
+        exp = df2.fillna(0).add(df1)
+        res = df2.add(df1, fill_value=0)
+        tm.assert_sp_frame_equal(res, exp)
```

Comment:
@jreback if I misunderstand, please tell me.

for sparse dataframe:
1. `fillna(x)`: convert NaN entries to `x`.
2. In `add(a, fill_value=x)`: `fill_value`, convert sparse entries to `x`.